### DISCUSSION TITLE
How to structure Google OR-Tools MPSolver to assign multiple tasks per worker?

### DISCUSSION THREAD

**Discussion by michaeldouble07** [google/or-tools#3130] [Linear Solver questions]
I'm following the MPSolver java examples in an attempt to assign two tasks per worker. A worker must perform two tasks or no tasks, a task can be assigned to only one worker. A matrix indicates the tasks that a worker can perform. The goal is to maximize the number of worker-task combinations. Some workers and tasks may not get assigned. The examples cover single worker-task assignments, but I don't follow how to structure MPSolver constraints to associate two tasks per worker.

Subsequently, I need to minimize the amount of time each worker spends. The time worked is associated with individual workers.

link:[https://stackoverflow.com/questions/71118139/how-to-structure-google-or-tools-mpsolver-to-assign-multiple-tasks-per-worker](https://stackoverflow.com/questions/71118139/how-to-structure-google-or-tools-mpsolver-to-assign-multiple-tasks-per-worker)

**Comment by lperron** [comment]
please provide a minimal runnable with some data. The question is too abstract as is.

Furthermore, what is the objective exactly ? minimize the sum of worker times ?

Assigning 0 or 2 tasks is trivial. `sum(tasks) == 2 * worker_is_active`, when `worker_is_active` is a Boolean variable.
